command/agent/command.go

```diff
@@ -470,14 +465,37 @@ func (cmd *Command) checkpointResults(results *checkpoint.CheckResponse, err err
 	}
 }
 
+func (cmd *Command) startupUpdateCheck(config *Config) {
+	version := config.Version
+	if config.VersionPrerelease != "" {
+		version += fmt.Sprintf("-%s", config.VersionPrerelease)
+	}
+	updateParams := &checkpoint.CheckParams{
+		Product: "consul",
+		Version: version,
+	}
+	if !config.DisableAnonymousSignature {
+		updateParams.SignatureFile = filepath.Join(config.DataDir, "checkpoint-signature")
+	}
+
+	// Schedule a periodic check with expected interval of 24 hours
+	checkpoint.CheckInterval(updateParams, 24*time.Hour, cmd.checkpointResults)
+
+	// Do an immediate check within the next 30 seconds
+	go func() {
+		time.Sleep(lib.RandomStagger(30 * time.Second))
```

Comment:
This should probably exit when the shutdown channel is closed (you didn't write this, but since you are in here).